app/controllers/organizations_controller.rb

```diff
@@ -0,0 +1,59 @@
+class OrganizationsController < ApplicationController
+  before_action :redirect_to_root, unless: :logged_in?
+
+  before_action :verify_organization_ownership, except: [:new, :create]
+  before_action :set_organization,              except: [:new, :create]
+
+  def new
+    @organization               = Organization.new
+    @users_github_organizations = current_user.github_client.users_organizations.
+                                  collect { |org| [org.login, [org.login, org.id]] }
+  end
+
+  def create
+    org               = JSON.parse(params[:org])
+    login, github_id  = org.first, org.last
+
+    if Organization.where(github_id: github_id).present?
+      redirect_to new_organization_path, alert: 'Classroom has already been added'
+    else
+      if current_user.github_client.is_organization_admin?(login)
```

Comment:
I recommend using `github_id` instead of login whenever you're doing user actions. That way, there's no chance of an error introduced by a username change.